File: apps/android/app/src/main/java/com/mishran/app/ui/checkout/components/PaymentMethodPicker.kt
// apps/android/app/src/main/java/com/mishran/app/ui/checkout/components/PaymentMethodPicker.kt — Task 10.2.
//
// Payment channel selection. This records the user's preference only — the
// Razorpay sheet (Task 10.3) owns method selection outright: the mobile SDK
// exposes no UPI preselect (prefill.method documents 'card' only, Android
// integration docs 2026-08-18), so the sheet always opens with every method
// available. Android's sheet lists installed UPI apps (GPay / PhonePe /
// BHIM) for one-tap inside its UPI tab — the UPI chip's label (B15) says
// so; the chip never deep-links a specific app itself.
package com.mishran.app.ui.checkout.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.material3.FilterChip
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.mishran.app.R
import com.mishran.app.ui.checkout.PaymentMethod

@OptIn(ExperimentalLayoutApi::class)
@Composable
fun PaymentMethodPicker(
    selected: PaymentMethod,
    onSelect: (PaymentMethod) -> Unit,
) {
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(
            stringResource(R.string.checkout_payment_title),
            style = MaterialTheme.typography.titleMedium,
            modifier = Modifier.semantics { heading() },
        )
        FlowRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            PaymentMethod.entries.forEach { method ->
                FilterChip(
                    selected = method == selected,
                    onClick = { onSelect(method) },
                    label = { Text(stringResource(method.labelRes)) },
                )
            }
        }
    }
}

File: apps/ios/Mishran/UI/Checkout/PaymentMethodPicker.swift
// PaymentMethodPicker.swift — Task 17.2 (Mishran Mobile Apps v1).
// Razorpay only in v1 (no COD/EMI per plan).
// B15 note: the Razorpay sheet owns method selection outright — the iOS
// SDK has no UPI preselect (and iOS is exempt from the 2026 UPI-collect
// deprecation, so the sheet's UPI tab keeps working). The label carries
// the UPI-first emphasis; no per-app deep links are promised.
import SwiftUI

struct PaymentMethodPicker: View {
    @Binding var selection: CheckoutViewModel.PaymentMethod?

    var body: some View {
        Picker(L("checkout.payment.title"), selection: $selection) {
            ForEach(CheckoutViewModel.PaymentMethod.allCases) { method in
                Text(method.displayName).tag(CheckoutViewModel.PaymentMethod?.some(method))
            }
        }
        .pickerStyle(.navigationLink)
        .accessibilityLabel(L("checkout.payment.title"))
    }
}
